src/Flex.php

```diff
@@ -757,18 +757,33 @@ public function fetchRecipes(array $operations): array
             }
 
             $noRecipe = !isset($manifests[$name]) || (isset($manifests[$name]['not_installable']) && $manifests[$name]['not_installable']);
-            if ($noRecipe && 'symfony-bundle' === $package->getType()) {
-                $manifest = [];
-                $bundle = new SymfonyBundle($this->composer, $package, $job);
-                if (null === $devPackages) {
-                    $devPackages = array_column($this->composer->getLocker()->getLockData()['packages-dev'], 'name');
+            if ($noRecipe) {
+                $bundles = [];
+
+                $extra = $package->getExtra();
+                if (isset($extra['symfony']['bundles'])) {
+                    $extraBundles = (array) $extra['symfony']['bundles'];
+                    foreach ($extraBundles as $bundleClass => $envs) {
+                        $bundles[$bundleClass] = (array) $envs;
+                    }
                 }
-                $envs = \in_array($name, $devPackages) ? ['dev', 'test'] : ['all'];
-                foreach ($bundle->getClassNames() as $class) {
-                    $manifest['manifest']['bundles'][$class] = $envs;
+
+                if (0 === \count($bundles)) {
+                    if (null === $devPackages) {
+                        $devPackages = array_column($this->composer->getLocker()->getLockData()['packages-dev'], 'name');
+                    }
+                    $envs = \in_array($name, $devPackages) ? ['dev', 'test'] : ['all'];
+                    $bundle = new SymfonyBundle($this->composer, $package, $job);
+                    foreach ($bundle->getClassNames() as $bundleClass) {
+                        $bundles[$bundleClass] = $envs;
+                    }
                 }
-                if ($manifest) {
-                    $manifest['origin'] = sprintf('%s:%s@auto-generated recipe', $name, $package->getPrettyVersion());
+
+                if (\count($bundles)) {
```

Comment:
`if ($bundles) {` (count is useless)

Reply:
https://github.com/symfony/flex/pull/612/commits/390179eec930c4a94a41755c5a5f4868b544cdea